ui/components/app/confirmations/alerts/alert-modal/alert-modal.tsx

```diff
@@ -225,20 +278,37 @@ export function AlertModal({
           borderWidth={1}
           display={headerStartAccessory ? Display.InlineFlex : Display.Block}
         />
-        <AlertHeader selectedAlert={selectedAlert} />
+        <AlertHeader
+          selectedAlert={selectedAlert}
+          customAlertTitle={customAlertTitle}
+        />
         <ModalBody>
-          <AlertDetails selectedAlert={selectedAlert} />
-          <AcknowledgeCheckbox
+          <AlertDetails
             selectedAlert={selectedAlert}
-            isConfirmed={isConfirmed}
-            setAlertConfirmed={setAlertConfirmed}
+            customAlertDetails={customAlertDetails}
           />
+          {customAcknowledgeCheckbox ?? (
+            <AcknowledgeCheckboxBase
+              selectedAlert={selectedAlert}
+              isConfirmed={isConfirmed}
+              onCheckboxClick={handleCheckboxClick}
+            />
+          )}
         </ModalBody>
         <ModalFooter>
-          <AcknowledgeButton
-            onAcknowledgeClick={onAcknowledgeClick}
-            isConfirmed={isConfirmed}
-          />
+          <Box
+            display={Display.Flex}
+            flexDirection={FlexDirection.Column}
+            gap={4}
+            width={BlockSize.Full}
+          >
+            {customAcknowledgeButton ?? (
+              <AcknowledgeButton
+                onAcknowledgeClick={onAcknowledgeClick}
+                isConfirmed={isConfirmed}
+              />
+            )}
+          </Box>
         </ModalFooter>
       </ModalContent>
```

Comment:
Q: will we add UI support for "Security advice by Blockaid" in this PR or a future iteration? 
![CleanShot 2024-05-21 at 00 57 35@2x](https://github.com/MetaMask/metamask-extension/assets/20778143/4f7c49e0-dc3d-4bf2-b925-03adf19acf6b)